src/api/voting.ts

```diff
@@ -565,21 +626,41 @@ export class VotingApi {
      * Retrieves the cumulative weight that has been casted in votes for the given process ID.
      * @param processId
      * @param gateway
+     * @param checkArchive
      */
-    static getResultsWeight(processId: string, gateway: IGateway | IGatewayPool): Promise<BigNumber> {
+    static getResultsWeight(processId: string, gateway: IGateway | IGatewayPool, checkArchive: boolean = false): Promise<BigNumber> {
         if (!processId) return Promise.reject(new Error("Empty process ID"))
         else if (!gateway) return Promise.reject(new Error("Invalid Gateway object"))
 
         return gateway.sendRequest({ method: "getResultsWeight", processId })
+            .catch((error) => {
+                const message = error.message ? "Could not retrieve the results weight: " + error.message : "Could not retrieve the results weight"
+                if (checkArchive && message.includes("No data found for this key")) {
+                    return this.getArchiveResultsWeight(processId, gateway, message)
+                }
+                throw new Error(message)
+            })
             .then((response) => {
                 if (response.weight < 0) throw new Error("The weight value is not valid")
                 else if (typeof response.weight !== 'string' && !BigNumber.isBigNumber(response.weight)) throw new Error("The weight value is not valid")
 
                 return BigNumber.from(response.weight)
             })
-            .catch((error) => {
-                const message = error.message ? "Could not retrieve the results weight: " + error.message : "Could not retrieve the results weight"
-                throw new Error(message)
+    }
+
+    /**
+     * Retrieves the archive cumulative weight that has been casted in votes for the given process ID.
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    private static getArchiveResultsWeight(processId: string, gateway: IGateway | IGatewayPool, errorMessage: string): Promise<DVoteGatewayResponseBody> {
```

Comment:
Same as before:
- Name
- Return plain data, don't expect to match someone else's schemas
- Needed here?